modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ComponentMessageProcessorObjectFactory.java

```diff
@@ -82,7 +88,13 @@ public P doGetObject() {
                                                                              nestedChain)));
 
       // For MULE-18771 we need access to the chain's location to create a new event and sdk context
-      nestedChain.setAnnotations(this.getAnnotations());
+      // Update: For W-15158118, to avoid issues with the registration of the chain in the component locator, we are not
+      // adding the location as the location key but as another annotation.
+      Map<QName, Object> annotations = new HashMap<>(this.getAnnotations());
+      ComponentLocation chainLocation = ((DefaultComponentLocation) annotations.get(LOCATION_KEY));
```

Comment:
Why cast? Also, isn't this the same as just calling this.getLocation()?

Reply:
No, because we removed the location from the annotations previously (that was the point of this)